### ISSUE TITLE
Node-mailjet causing requests library "got" to fail?

### ISSUE THREAD

**Issue by ripperdoc** [mailjet/mailjet-apiv3-nodejs#128]
- Node.js version: 14.8.0
 - OS & version: MacOs 10.14.5

The packages `got` (popular for HTTP requests) and `node-mailjet` somehow interact to create a bug that throws on all `got`-requests. I don't understand what's going on under the hood so I report this potential bug both here and over at `got`. This ONLY happens if I require `node-mailjet`, otherwise got works as expected, so I suspect the issue originates from mailjet.

`got("https://google.com")` throws `RequestError: connect ECONNREFUSED`. Full stack trace:
```
RequestError: connect ECONNREFUSED 127.0.0.1:443
    at ClientRequest.<anonymous> (/Users/martin/dev/src/test/node_modules/got/dist/source/core/index.js:953:111)
    at Object.onceWrapper (events.js:421:26)
    at ClientRequest.emit (events.js:326:22)
    at ClientRequest.origin.emit (/Users/martin/dev/src/test/node_modules/@szmarczak/http-timer/dist/source/index.js:39:20)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:314:20)
    at emitErrorNT (internal/streams/destroy.js:100:8)
    at emitErrorCloseNT (internal/streams/destroy.js:68:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1144:16) {
  code: 'ECONNREFUSED',
  timings: {
    start: 1603894700629,
    socket: 1603894700629,
    lookup: 1603894700630,
    connect: undefined,
    secureConnect: undefined,
    upload: undefined,
    response: undefined,
    end: undefined,
    error: 1603894700630,
    abort: undefined,
    phases: {
      wait: 0,
      dns: 1,
      tcp: undefined,
      tls: undefined,
      request: undefined,
      firstByte: undefined,
      download: undefined,
      total: 1
    }
  }
}
```

#### Code to reproduce
Setup a fresh npm package with the following code and ran:
```js
// index.js
async function test() {
    const got = require('got');
    const mailjet = require("node-mailjet");
    try {
        console.log(await got("https://google.com"));
    } catch (e) {
        console.error(e);
    }}

test();
// package.json
{
  "name": "test",
  "main": "index.js",
  "dependencies": {
    "got": "11.8.0",
    "node-mailjet": "3.3.1"
  }
}


```
If I comment out the mailjet require, the code works as expected, so the likely culprit is node-mailjet, but perhaps it has some impact what's happening in got's code as well.

**Comment by Giotino** [comment]
The mailjet team should update the package `superagent-proxy` and it should be fine.

Explanation:
`node-mailjet` -> `superagent-proxy@^1.0.2(currently 1.0.3)` -> `proxy-agent@2(currently 2.3.1)` -> `agent-base@^4.2.0(currently 4.3.0)`

`agent-base@4` was patching Node.js methods, which was incompatible with Got.

https://github.com/sindresorhus/got/issues/1018

**Comment by katafractari** [comment]
Like @Giotino said, all that had to be done was updating superagent and superagent-proxy dependencies.

**Comment by iSuslov** [comment]
You are welcome to use this fork as a dependency.

"node-mailjet": "https://github.com/iSuslov/mailjet-apiv3-nodejs.git"

**Comment by andrii-yelis** [comment]
superagent has been updated.

**Comment by rockebee** [comment]
@andrii-yelis, I might miss something, but why is this issue closed? The respective PR #134 is still open and _not_ merged to master nor published anywhere ...?! (Neither is PR #147, which would only address superagent-proxy.)
In any case, actual issue with https://github.com/TooTallNate/node-agent-base/issues/35 is not solved, so IMHO it'd be great to re-open this issue and close only once the PR got merged and released.
